src/signed.rs

```diff
@@ -490,10 +490,8 @@ macro_rules! int_impl {
                 #[inline]
                 #[must_use = "this returns the result of the operation, without modifying the original"]
                 pub const fn wrapping_add(self, rhs: Self) -> Self {
-                    let sum = self.value.wrapping_add(rhs.value);
-                    Self {
-                        value: (sum << Self::UNUSED_BITS) >> Self::UNUSED_BITS,
-                    }
+                    let (value, _overflow) = self.overflowing_add(rhs);
```

Comment:
do we trust the optimizer it will produce equally good code? at least for debug builds this might be a little worse

Reply:
Hmm, it does indeed look like this produces worse code for debug builds, though optimized builds aren't affected. The slight amount of deduplication probably isn't worth it, I'll revert it and only keep the `self.value` -> `self.value()` part.